microsoft-edge/progressive-web-apps-chromium/how-to/digital-goods-api.md

```diff
@@ -0,0 +1,213 @@
+
+See [Payment Request API](https://www.w3.org/TR/payment-request/) at W3C.
+
+
+<!-- ====================================================================== -->
+## Checking whether the Digital Goods API is available
+
+To detect whether you've correctly enabled the API on your website, check for the `getDigitalGoodsService` method in the window object:
+
+```javascript
+if ('getDigitalGoodsService' in window) {
+  // Digital Goods API is supported!
+} else {
+  console.log('DigitalGoodsService is not available.');
+  // Use other payment method
+}
+```
+
+
+<!-- ====================================================================== -->
+## Connecting to the Microsoft Store Billing service (`getDigitalGoodsService` method)
+
+Use the `getDigitalGoodsService` method to connect to the Microsoft Store Billing service.
+
+The Digital Goods API was designed to be compatible with various browsers and digital stores, similar to how the Payment Request API is browser-agnostic and can be used with different payment providers.  To retrieve an instance of the service for Microsoft Store Billing, pass the string `"https://store.microsoft.com/billing"` as the payment method to the `getDigitalGoodsService` method.
+
+If the method throws an error, the Microsoft Store Billing payment method is not available (such as when the user is accessing your PWA through the browser).  Alternatively, consider providing a different payment method for transactions.
+
+```javascript
+if (window.getDigitalGoodsService === undefined) {
+  // Digital Goods API is not supported in this context.
+  return;
+}
+try {
+  const digitalGoodsService = await window.getDigitalGoodsService("https://store.microsoft.com/billing");
+  // Use the service here.
+  ...
+} catch (error) {
+  // Our preferred service provider is not available.
+  // Use a normal web-based payment flow.
+  return;
+}
+```
+
+
+<!-- ====================================================================== -->
+## Querying item details (`getDetails` method)
+
+Use the `getDetails` method to query item details.
+
+After connecting the Digital Goods service to Microsoft Store, you can use the API to access product and purchase information.  The `getDetails` method lets you get information about the items you’ve set up in the Partner Center.  Display information such as the product title, description, and price in your app UI, so the user knows what's available for purchase.
+
+The `getDetails` method takes a list of item IDs, which correspond to the product IDs of the in-app products and subscriptions you created in the Partner Center.
+
+```javascript
+details = await digitalGoodsService.getDetails(['shiny_sword', 'gem', 'monthly_subscription']);
+for (item of details) {
+  const priceStr = new Intl.NumberFormat(
+      locale,
+      {style: 'currency', currency: item.price.currency}
+    ).format(item.price.value);
+  AddShopMenuItem(item.itemId, item.title, priceStr, item.description);
+}
```

Comment:
```suggestion
const itemDetails = await digitalGoodsService.getDetails(['shiny_sword', 'gem', 'monthly_subscription']);

for (item of itemDetails) {
  const priceStr = new Intl.NumberFormat(
      locale,
      {style: 'currency', currency: item.price.currency}
    ).format(item.price.value);

  // Do something with the item's data, such as displaying it in the PWA's UI.
  displayProductItem(item.itemId, item.title, priceStr, item.description);
}
```

Changing `details` to `itemDetails` since that's what's used in the next paragraph.